### PR TITLE
Add spring boot demo application version 2.7.18

### PR DESCRIPTION
**Pull request by KiKoS0**
- https://start.spring.io/ by default supports spring boot version 3.x. 3.x is not compatible with Java 8, so we downgraded the generated  app to 2.7.18 which is compatible. https://www.marcobehler.com/guides/spring-and-spring-boot-versions
- We changed the toolchain for `inngest-core` to Java 8 so that Java 8 apps can consume it. This meant we couldn't use Java httpclient which [was introduced in Java 11](https://openjdk.org/groups/net/httpclient/intro.html).
- We evaluated a couple of Http clients and went with OkHttp because its API resembles the Java 11 Http client the most (Other clients were Google Http client and Apache Http client)
- The current state is that the app syncs correctly to the dev server but when a function is called it fails on steps execution because of two issues that we'll be looking into next:
```
  Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Handler dispatch failed; nested exception is java.lang.NoClassDefFoundError: kotlin/enums/EnumEntriesKt] with root cause

java.lang.ClassNotFoundException: kotlin.enums.EnumEntriesKt
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641) ~[na:na]
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[na:na]
	at com.inngest.OpCode.<clinit>(Function.kt:26) ~[main/:na]
	at com.inngest.InngestFunction.call(Function.kt:124) ~[main/:na]
	at com.inngest.CommHandler.callFunction(Comm.kt:85) ~[main/:na]
	at com.inngest.springbootdemo.InngestController.handleRequest(InngestController.java:40) ~[main/:na]
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:578) ~[na:na]
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205) 
```
- The second issue is that reified inline functions [cannot be called from Java](https://stackoverflow.com/a/42742119) and `step.run` currently is. 
https://github.com/inngest/inngest-kt/blob/e9a6a15498d742a775a8a5bbd107cada10d9bb31/inngest-core/src/main/kotlin/com/inngest/Step.kt#L28

### CODE REVIEW

**Review by albertchae — APPROVED** [review]
paired on this together, there are still some outstanding issues as detailed in the PR description but we figured out enough of the Java 8 compilation that we thought it was worth merging something now

File: inngest-spring-boot-demo/src/main/java/com/inngest/springbootdemo/InngestSingleton.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public Result(@JsonProperty("sum") int sum) {
+        this.sum = sum;
+    }
+
+    public int getSum() {
+        return sum;
+    }
+}
+
+// NOTE: We probably don't need this singleton anymore
```
Comment (darwin67) [inline inngest-spring-boot-demo/src/main/java/com/inngest/springbootdemo/InngestSingleton.java:25]: The current SDK standards are to no expose any of the handlers to the users, and they can either,
- mount our route onto their HTTP controller
- directly create the HTTP endpoint for them

I'd imagine Java will be the former, so the singleton shouldn't exist and only the controller should be there.

**Review by darwin67 — APPROVED** [review]
looks good.
could you add the command you use to run spring boot server before you merge?

**Comment by albertchae** [comment]
> looks good. could you add the command you use to run spring boot server before you merge?

we need to figure out this out still 😅 , for now we were just running via IntelliJ

**Pull request merged**